pages/search.js

```diff
@@ -44,18 +44,18 @@ export default function Search({ data }) {
   const [filteredUsers, setFilteredUsers] = useState([]);
   const [inputValue, setInputValue] = useState(username || keyword || "");
 
-  let results = [];
 
   useEffect(() => {
     if (username) {
       setNotFound(username);
     }
   }, [username]);
 
-  const filterData = (value) => {
+  const filterData = useCallback((value) => {
```

Comment:
By wrapping `filterData` in `useCallback`, we ensure that it's the same function between the re-renders(until dependencies change).